extensions/wikia/AttributeServiceMWApi/AttributeServiceMWApiController.class.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+class AttributeServiceMWApiController extends WikiaController {
+
+    /**
+     * Helper function for the attribute service to clear user MediaWiki.
+     * This is called when a client other than MW updates an attribute
+     * using the attribute service to clear the User cache.
+     */
+    public function purgeUserCache() {
+
+        $userId = $this->request->getVal( 'userId' );
+        $signature = $this->request->getVal( 'signature' );
+
+       if ( !$this->assertValidParameters( $userId, $signature ) ) {
+           return;
```

Comment:
It actually sets the error response code [here](https://github.com/Wikia/app/pull/8329/files#diff-341ac26a93d9bf763b1539072665fa4cR39). But, it's not super obviously that asserting here has the side effect of altering the response. I'll move the `setErrorResponse` call higher up to make that more clear.